solidity/core/contracts/validator-manager/OutboxMultisigValidatorManager.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.6.11;
+pragma abicoder v2;
+
+// ============ Internal Imports ============
+import {IOutbox} from "../../interfaces/IOutbox.sol";
+import {MultisigValidatorManager} from "./MultisigValidatorManager.sol";
+
+/**
+ * @title OutboxMultisigValidatorManager
+ * @notice Verifies if an improper checkpoint has been signed by a quorum of
+ * validators and reports it to an Outbox.
+ */
+contract OutboxMultisigValidatorManager is MultisigValidatorManager {
+    // ============ Events ============
+
+    /**
+     * @notice Emitted when proof of an improper checkpoint is submitted.
+     * @dev Observers of this event should filter by the outbox address.
+     * @param outbox The outbox.
+     * @param root Root of the improper checkpoint.
+     * @param index Index of the improper checkpoint.
+     * @param signatures A quorum of signatures on the improper checkpoint.
+     * May include non-validator signatures.
+     */
+    event ImproperCheckpoint(
+        address indexed outbox,
+        bytes32 indexed root,
+        uint256 index,
+        bytes[] signatures
+    );
+
+    // ============ Constructor ============
+
+    /**
+     * @param _localDomain The local domain.
+     * @param _validatorSet The set of validator addresses.
+     * @param _quorumThreshold The quorum threshold. Must be greater than or equal
+     * to the length of _validatorSet.
+     */
+    // solhint-disable-next-line no-empty-blocks
+    constructor(
+        uint32 _localDomain,
+        address[] memory _validatorSet,
+        uint256 _quorumThreshold
+    ) MultisigValidatorManager(_localDomain, _validatorSet, _quorumThreshold) {}
+
+    // ============ External Functions ============
+
+    /**
+     * @notice Determines if a quorum of validators have signed an improper checkpoint,
+     * failing the Outbox if so.
+     * @dev Improper checkpoints signed by individual validators are not handled to prevent
+     * a single byzantine validator from failing the Outbox.
+     * @param _outbox The outbox.
+     * @param _root The merkle root of the checkpoint.
+     * @param _index The index of the checkpoint.
+     * @param _signatures Signatures over the checkpoint to be checked for a validator
+     * quorum. Must be sorted in ascending order by signer address.
+     */
+    function improperCheckpoint(
+        IOutbox _outbox,
+        bytes32 _root,
+        uint256 _index,
+        bytes[] calldata _signatures
+    ) external {
+        require(isQuorum(_root, _index, _signatures), "!quorum");
+        require(!_outbox.isCheckpoint(_root, _index), "!improper checkpoint");
```

Comment:
not sure I love these "!error" messages
the boolean implications are confusing

Reply:
It took me a little time to get used to it too. Wdyt of keeping with the pattern for now and we can consider moving the repo's revert strings separately?
Or do you find these strings particularly confusing